etc/kayobe/ansible/tools/get-cloud-facts.yml

```
@@ -1,13 +1,68 @@
 ---
-- name: Gather Cloud Facts
+- name: Set facts for control host
+  hosts: localhost
+  gather_facts: true
+  tasks:
+    - name: Check if ngs enabled
+      ansible.builtin.set_fact:
+        ngs_enabled: >-
+          {{ 'genericswitch' in kolla_neutron_ml2_mechanism_drivers | default(false) }}
+      when: kolla_neutron_ml2_mechanism_drivers
+
+- name: Set facts for seed node
+  hosts: seed
+  gather_facts: true
+  tasks:
+    - name: Is there a dedicated seed node?
+      ansible.builtin.set_fact:
+        seed_node_is_vm: "{{ ansible_facts.virtualization_role == 'guest' }}"
+
+- name: Set facts for all hosts
+  hosts: all
+  gather_facts: true
+  tasks:
+    - name: Pinging stackhpc.com to check internet connectivity
+      ansible.builtin.command: "ping stackhpc.com -c 3"
+      register: successful_ping
+      timeout: 5
+      ignore_errors: true
+
+    - name: Get kernel version
+      ansible.builtin.command: "uname -r"
+      register: kernel_version
+
+    - name: Check if Docker is installed
+      ansible.builtin.command: "which docker"
+      register: docker_check
+      failed_when: docker_check.rc not in [0]
+      ignore_errors: true
+
+    - name: Get running contianers
+      ansible.builtin.command: "docker ps --format {% raw %}'{{.Names}}'{% endraw %}"
+      become: true
+      register: containers_list
+      when: not docker_check.failed
+
+    - name: Set facts
+      ansible.builtin.set_fact:
+        internet_connectivity: "{{ not successful_ping.failed }}"
+        distribution: "{{ ansible_facts.distribution }}"
+        distribution_release: "{{ ansible_facts.distribution_release }}"
+        hugepages_enabled: "{{ 'hugepages' in ansible_facts.cmdline }}"
+
+- name: Write cloud facts file
   hosts: localhost
   gather_facts: true
   tasks:
     - name: Write facts to file
       vars:
         cloud_facts:
-          ansible_control_host_distribution: "{{ ansible_facts.distribution }}"
-          ansible_control_host_distribution_release: "{{ ansible_facts.distribution_release }}"
+          internet_connectivity: "{{ hostvars | group_hostvars_by_var('internet_connectivity') }}"
+          seed_node_is_vm: "{{ hostvars[groups['seed'][0]]['seed_node_is_vm'] | default('N/A') if groups ['seed'] else 'N/A' }}"
+          distributions: "{{ hostvars | group_hostvars_by_var('distribution') }}"
+          dist_releases: "{{ hostvars | group_hostvars_by_var('distribution_release') }}"
+          kernel_versions: "{{ hostvars | group_hostvars_by_var('kernel_version','stdout') }}"
+          deployed_containers: "{{ hostvars | get_hostvars_by_host('containers_list', 'stdout_lines') | default({}) }}"
```

Couple of minor spacing issues here (see linter warnings)
```suggestion
          seed_node_is_vm: "{{ hostvars[groups['seed'][0]]['seed_node_is_vm'] | default('N/A') if groups['seed'] else 'N/A' }}"
          distributions: "{{ hostvars | group_hostvars_by_var('distribution') }}"
          dist_releases: "{{ hostvars | group_hostvars_by_var('distribution_release') }}"
          kernel_versions: "{{ hostvars | group_hostvars_by_var('kernel_version', 'stdout') }}"
          deployed_containers: "{{ hostvars | get_hostvars_by_host('containers_list', 'stdout_lines') | default({}) }}"
```